java/create-and-save-map/src/main/java/com/esri/arcgisruntime/sample/createandsavemap/MainActivity.java

```diff
@@ -249,22 +284,19 @@ private void showSaveMapDialog() {
    * @param description of the map
    */
   private void saveMap(String title, Iterable<String> tags, String description) {
-    // create a portal to arcgis
-    mPortal = new Portal("https://www.arcgis.com", true);
     mPortal.addDoneLoadingListener(() -> {
       if (mPortal.getLoadStatus() == LoadStatus.LOADED) {
         // call save as async and pass portal info, as well as details of the map including title, tags and description
         ListenableFuture<PortalItem> saveAsAsyncFuture = mMapView.getMap()
-            .saveAsAsync(mPortal, null, title, tags, description, null, true);
+            .saveAsAsync(mPortal, mPortalFolders.get(mFolderSpinner.getSelectedItemPosition()), title, tags, description, null, true);
         saveAsAsyncFuture.addDoneListener(
             () -> Toast.makeText(this, "Map saved to portal!", Toast.LENGTH_LONG).show());
       } else {
-        String error = "Error loading portal: " + mPortal.getLoadError().getMessage();
+        String error = "Can't save map, portal not loaded: " + mPortal.getLoadError().getMessage();
```

Comment:
with the addition of loading the portal code on line 228, there is repeated code here that could be tidied up to save calling the portal's `addDoneLoadingListener` twice. 

Since the portal has already been checked as loaded when the save map dialog opens, it should be possible and safe to just call the following in this method: the only way a user could get to this point is if the portal has loaded successfully in `showSaveMapDialog`

```java

    ListenableFuture<PortalItem> saveAsAsyncFuture = mMapView.getMap()
            .saveAsAsync(mPortal, mPortalFolders.get(mFolderSpinner.getSelectedItemPosition()), title, tags, description, null, true);
        saveAsAsyncFuture.addDoneListener(
            () -> Toast.makeText(this, "Map saved to portal!", Toast.LENGTH_LONG).show());

```

Reply:
Good point! Changed that to remove checks